### ISSUE TITLE
Web App Schedule Recording Save Button Incorrectly Disabled

### ISSUE THREAD

**Issue by mymatenige** [MythTV/mythtv#897]
* **Platform**:
```
Fedora 40
Linux pvr.grufty.co.uk 6.8.8-300.fc40.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Apr 27 17:53:31 UTC 2024 x86_64 GNU/Linux
```
* **MythTV version**:
`fixes/34`
* **Package version**:
N/A (Source)
* **Component**:
`Web App`

### What steps will reproduce the bug?
The following does not work:

Go to Upcoming tab
Find a program that "Will Record"
Clip on the paperclip
Change the Schedule Type to "Do not record this showing"
Click Save
Click Close
FInd the same program now showing as "Dont Record"
Clip on the paperclip
Change the Schedule Type to "Record this showing with normal options"
The Save button is disabled, it should be enabled

The following works fine (notice the lack of Close):

Go to Upcoming tab
Find a program that "Will Record"
Clip on the paperclip
Change the Schedule Type to "Do not record this showing"
Click Save
Change the Schedule Type to "Record this showing with normal options"
Click Save

### How often does it reproduce? Is there a required condition?
Consistent reproduction.

### What is the expected behaviour?
The Save button is not disabled.

### What do you see instead?
The Save button is disabled.

### Additional information

<img width="549" alt="Screenshot 2024-05-08 at 20 45 01" src="https://github.com/MythTV/mythtv/assets/679665/eb81accd-bef3-481e-afe7-308e608a860a">

**Comment by bennettpeter** [comment]
I am unable to reproduce this.

> Go to Upcoming tab
> Find a program that "Will Record"
> Clip on the paperclip
> Change the Schedule Type to "Do not record this showing"
> Click Save
> Click Close
> FInd the same program now showing as "Dont Record"
> 
The program does not show in upcoming recordings as "Don't Record". The showing that is no longer to be recorded is no longer on the list. 

Perhaps this is what is happening:
The program shows in the same time as before that it is now recording on another channel as "Will Record". If I click the paper clip on that "Will Record" entry, and select "Record this showing with normal options", save is disabled. This is intentional, because recording with normal options happens anyway, on that alternate channel, if I do not save.

If you have changed a recording to "Do not record this showing", the way to undo that action is to go into the "Recording Rules" tab, find the entry with "Do Not Record", and either delete it or click the paperclip and select "Record with Normal Options".

**Comment by bennettpeter** [comment]
I found a situation where the show is in fact in the upcoming list with "Dont Record" . This is when I set "Don't Record" using a different API. I will sort out the mess.

**Comment by bennettpeter** [comment]
I found the problem. It is a simple fix. I will commit it together with the fix for #899

**Comment by mymatenige** [comment]
> I found the problem. It is a simple fix. I will commit it together with the fix for #899

Great, thanks Peter.  I've been trying to look to find to replicate it but not had a chance yet.  Appreciate you looking into it.

**Comment by bennettpeter** [comment]
Fixed in b1203fe77e and f593133c74
